src/pages-and-resources/discussions/DiscussionsSettings.jsx

```diff
@@ -24,80 +21,81 @@ import { fetchApps } from './data/thunks';
+
+  useEffect(() => {
+    setCurrentStep(appId ? SETTINGS_STEP : SELECTION_STEP);
+  }, [appId]);
 
   const handleClose = useCallback(() => {
     history.push(pagesAndResourcesPath);
-  }, [courseId]);
+  }, [pagesAndResourcesPath]);
 
   const handleBack = useCallback(() => {
-    if (isFirstStep) {
-      history.push(pagesAndResourcesPath);
-    } else {
-      history.push(discussionsPath);
-    }
-  }, [discussionsPath, isFirstStep, pagesAndResourcesPath]);
+    history.push(discussionsPath);
+  }, [discussionsPath]);
 
   return (
     <DiscussionsProvider path={discussionsPath}>
-      <FullScreenModal title={intl.formatMessage(messages.configure)} onClose={handleClose}>
-        <FullScreenModal.Header title={intl.formatMessage(messages.configure)} />
-        <FullScreenModal.Body className="d-flex flex-column">
-          <AppConfigForm.Provider>
-            <Stepper className="h-100">
-              <Stepper.Header steps={steps} />
-              <Stepper.Body className="bg-light-200">
-                <Switch>
-                  <PageRoute exact path={`${path}`}>
-                    <AppList />
-                  </PageRoute>
-                  <PageRoute path={`${path}/configure/:appId`}>
-                    <AppConfigForm
-                      courseId={courseId}
-                    />
-                  </PageRoute>
-                </Switch>
-              </Stepper.Body>
-              <Stepper.Footer className={`d-flex ${isFirstStep ? 'justify-content-end' : 'justify-content-between'} align-items-center`}>
-                {!isFirstStep && (
-                  <Button variant="outline-primary" className="ml-2" onClick={handleBack}>
-                    {intl.formatMessage(messages.backButton)}
-                  </Button>
-                )}
-                {isFirstStep && (
+      <AppConfigForm.Provider>
+        <Stepper activeKey={currentStep}>
+          <FullscreenModal
+            className="bg-light-200"
+            title={intl.formatMessage(messages.configure)}
+            onClose={handleClose}
+            isOpen
+            footerNode={(
+              <>
+                <Stepper.ActionRow eventKey={SELECTION_STEP}>
                   <AppList.NextButton />
-                )}
-                {!isFirstStep && (
-                  <AppConfigForm.ApplyButton />
-                )}
-              </Stepper.Footer>
-            </Stepper>
-          </AppConfigForm.Provider>
-        </FullScreenModal.Body>
-      </FullScreenModal>
+                </Stepper.ActionRow>
+                <Stepper.ActionRow eventKey={SETTINGS_STEP}>
+                  <div className="d-flex w-100 justify-content-between">
+                    <Button
+                      aria-label={intl.formatMessage(messages.backButton)}
+                      variant="outline-primary"
+                      onClick={handleBack}
+                    >
+                      {intl.formatMessage(messages.backButton)}
```

Comment:
Do you want to separate them? and move the back button to the left?

Reply:
This is done - see line 68 above.  It puts them in a width 100% flexbox and with justify-content-between, which moves them to the left and right of the flexbox.

Reply:
For the future: there is an `ActionRow.Spacer` subcomponent (in this case `Stepper.ActionRow.Spacer`) which has a `flex-grow: 1 ` style assigned to it. This will achieve the same result as you have above:

```
<Stepper.ActionRow>
  <TheBackButton />
  <Stepper.ActionRow.Spacer />
  <TheNextButton />
</Stepper.ActionRow>
```